rainier-core/src/main/scala/com/stripe/rainier/core/Continuous.scala

```diff
@@ -141,6 +141,34 @@ object Exponential {
     standard.scale(Real.one / rate)
 }
 
+/**
+  * A Beta distribution with expectation `a/(a + b)` and variance `ab/((a + b)^2 (1 + a + b))`.
+  */
+object Beta {
+  def apply(a: Real, b: Real): Continuous = new Continuous {
+    def realLogDensity(p: Real): Real =
+      If(p < 0,
+         Real.zero.log,
+         If(p > 1,
+            Real.zero.log,
+            (a - 1) * p.log + (b - 1) * (1 - p).log - Combinatrics.beta(a, b)))
+
+    def param: RandomVariable[Real] = {
```

Comment:
The duplication with `Uniform` for `realLogDensity` and `param` bothers me a bit. Two possible solutions:

* define `Uniform.standard` as `Beta(1,1)`. If you did this, you'd be making uniform's logDensity slightly different from 0.0 because of the approximate gamma function; not sure if this ever matters in practice. Also, you'd probably want to special case the Beta.generator for the (1,1) case to just use `r.standardUniform` (but maybe you want to do that anyway). But there's a certain elegance to it.
* or, do something like defining Beta's `realLogDensity` as `Uniform.standard.logDensity(p) + betaDensity(p)`, and define `param` as `Uniform.standard.flatMap{u => RandomVariable(u, betaDensity(u))}`. Then put the ` `(a - 1) * p.log + (b - 1) * (1 - p).log - Combinatrics.beta(a, b)))` inside a private `betaDensity` method. This is maybe less elegant but more practical.